external-import/cofense/src/main.py

```diff
@@ -0,0 +1,315 @@
+import os
+import sys
+import time
+import traceback
+from datetime import datetime, timedelta
+from typing import Any
+
+import stix2
+import yaml
+from cofense_intelligence import CFIntelSync, CofenseIntegration, MalwareThreatReport
+from pycti import (
+    Incident,
+    OpenCTIConnectorHelper,
+    StixCoreRelationship,
+    get_config_variable,
+)
+
+config_file_path = os.getenv(
+    "OPENCTI_CONFIG", os.path.dirname(os.path.abspath(__file__)) + "/config.yml"
+)
+config = (
+    yaml.load(open(config_file_path), Loader=yaml.FullLoader)
+    if os.path.isfile(config_file_path)
+    else {}
+)
+helper = OpenCTIConnectorHelper(config)
+interval_sec = get_config_variable(
+    "COFENSEINTEL_INTERVAL", ["cofense", "interval"], config, True
+)
+user_token = get_config_variable(
+    "COFENSEINTEL_USER", ["cofense", "user"], config, False
+)
+user_pass = get_config_variable(
+    "COFENSEINTEL_PASSWORD", ["cofense", "password"], config, False
+)
+update_existing_data = get_config_variable(
+    "COFENSEINTEL_UPDATE", ["cofense", "update_existing_data"], config, False
+)
+
+BLOCK_TYPE = {
+    "URL": "url",
+    "Domain Name": "domain-name",
+    "IPv4 Address": "ipv4-addr",
+    "Email": "email-addr",
+}
+
+BLOCK_OBJ = {
+    "URL": stix2.URL,
+    "Domain Name": stix2.DomainName,
+    "IPv4 Address": stix2.IPv4Address,
+    "Email": stix2.EmailAddress,
+}
+
+
+class CofenseIntel(CofenseIntegration):
+
+    def _get_labels(self, mrti: MalwareThreatReport) -> list:
+        labels = []
+
+        labels.extend(f"family:{x}" for x in mrti.malware_families)
+        labels.extend(f"delivery:{x}" for x in mrti.delivery_mechs)
+        labels.extend(f"brand:{x}" for x in mrti.brands)
+
+        return labels
+
+    def _create_incident(
+        self,
+        mrti: MalwareThreatReport,
+        author_id: str,
+        mrti_labels: list,
+        created: datetime,
+    ) -> Incident:
+        incident = stix2.Incident(
+            id=Incident.generate_id(mrti.label, created),
+            name=mrti.label,
+            description=f"{mrti.label}\n{mrti.executive_summary}",
+            object_marking_refs=[stix2.TLP_RED],  # TODO: Is this correct?
```

Comment:
Is it the original intention to define "TLP_RED" for all “intrusions set” and related “relationships” ?